client/chess.ts

```diff
@@ -67,7 +69,7 @@ type MandatoryPromotionPredicate = (role: cg.Role, orig: cg.Key, dest: cg.Key, c
 const alwaysMandatory: MandatoryPromotionPredicate = () => true;
 
 function distanceBased(required: { [ letter: string ]: number }, boardHeight: number) {

```

Comment:
Actually every single function should have a return type. For this one it's `MandatoryPromotionPredicate`.